torch/utils/data/datapipes/iter/callable.py

```diff
@@ -227,3 +228,60 @@ def __init__(
                 # TODO(VitalyFedyunin): Validate passed dictionary
                 collate_fn = functools.partial(_collate_helper, conversion)
                 super().__init__(datapipe, fn=collate_fn)
+
+@functional_datapipe('drop')
+class DropperIterDataPipe(IterDataPipe):
+    r"""
+    Drop columns/elements in input DataPipe via its indices (functional name: ``drop``).
+
+    Args:
+        datapipe: MapDataPipe with columns to be dropped
+        indices: a single column index to be dropped or a list of indices
+
+    Example:
+        >>> from torchdata.datapipes.iter import IterableWrapper, ZipperMapDataPipe
+        >>> dp1 = IterableWrapper(range(5))
+        >>> dp2 = IterableWrapper(range(10, 15))
+        >>> dp = dp1.zip(dp2)
+        >>> list(dp)
+        [(0, 10), (1, 11), (2, 12), (3, 13), (4, 14)]
+        >>> drop_dp = dp.drop(1)
+        >>> list(drop_dp)
+        [(0), (1), (2), (3), (4)]
+    """
+    datapipe: IterDataPipe
+
+    def __init__(self,
+                 datapipe: IterDataPipe,
+                 indices: Union[int, List],
```

Comment:
As we will put `indices` into a set, the type of `indices` should be any `Hashable` object, List of `Hashable` or `set`.  `int` is not sufficient considered we would support any key for `dict`.